src/components/NavigationBar/index.jsx

```diff
@@ -1,47 +1,67 @@
-import React from "react";
+import { useEffect, useState } from "react";
 import PropTypes from "prop-types";
-import { Navbar, NavbarBrand, NavbarContent, NavbarItem, Link, Button } from "@nextui-org/react";
+import {
+  Navbar,
+  NavbarBrand,
+  NavbarContent,
+  NavbarItem,
+  Link,
+  Button,
+} from "@nextui-org/react";
 import "./style.css";
 import { useNavigate } from "react-router-dom";
 
-export const NavigationBar = ({isLoggedin}) => {
+export const NavigationBar = () => {
   const navigateTo = useNavigate();
-  
+  const [isLoggedin, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
+    const loginStatus = localStorage.getItem("loginStatus"); //this will always return a string so it needs to be coerced
+    setIsLoggedIn(loginStatus === "true");
+  }, []);
+
   return (
-    (isLoggedin)?
     <Navbar>
       <NavbarBrand>
-        <div className="logo-text cursor-pointer" onClick={() => navigateTo("/")}>Job Quest</div>
-      </NavbarBrand>
-
-      <NavbarContent className="hidden sm:flex gap-4" justify="center">
-        <NavbarItem className="nav-button mt-5">
-          <Link color="login" href="/login" className="login-text">Login</Link>
-        </NavbarItem>
-        <NavbarItem className="nav-button mt-5">
-          <Button as={Link} href="/login" className="signup-text">
-            Sign Up
-          </Button>
-        </NavbarItem>
-      </NavbarContent>
-      </Navbar>
-      :
-      <Navbar>
-      <NavbarBrand>
-        <div className="logo-text cursor-pointer" onClick={() => navigateTo("/")}>Job Quest</div>
+        <div
+          className="logo-text cursor-pointer"
+          onClick={() => navigateTo("/")}
+        >
+          Job Quest
+        </div>
       </NavbarBrand>
 
       <NavbarContent className="hidden sm:flex gap-4" justify="center">
-        <NavbarItem className="nav-button mt-5">
-          <Button as={Link} href="/login" className="signup-text">
-            Log Out
-          </Button>
-        </NavbarItem>
+        {isLoggedin ? (
```

Comment:
I now realize the flaw in my code in this section. Thanks for the correction @khairahscorner